index.bs

```diff
@@ -2112,53 +2124,59 @@ partial interface MLGraphBuilder {
+                    1. Let |filterOutputChannels| be |filterShape[1].
                     1. Let |filterHeight| be |filterShape|[2].
                     1. Let |filterWidth| be |filterShape|[3].
                 : {{MLConvTranspose2dFilterOperandLayout/"hwoi"}}
                 ::
                     1. Let |filterHeight| be |filterShape|[0].
                     1. Let |filterWidth| be |filterShape|[1].
+                    1. Let |filterOutputChannels| be |filterShape[2].
+                    1. Let |filterInputChannels| be |filterShape|[3].
                 : {{MLConvTranspose2dFilterOperandLayout/"ohwi"}}
                 ::
+                    1. Let |filterOutputChannels| be |filterShape[0].
                     1. Let |filterHeight| be |filterShape|[1].
                     1. Let |filterWidth| be |filterShape|[2].
+                    1. Let |filterInputChannels| be |filterShape|[3].
             </dl>
-        1. Let |outputSizes| be the result of [=MLGraphBuilder/calculating convtranspose2d output sizes=] given |inputHeight|, |inputWidth|, |filterHeight|, |filterWidth|, |options|.{{MLConvTranspose2dOptions/padding}}, |options|.{{MLConvTranspose2dOptions/strides}}, |options|.{{MLConvTranspose2dOptions/dilations}}, and |options|.{{MLConvTranspose2dOptions/outputPadding}}.
+        1. Let |outputChannels| be |filterOutputChannels| * |options|.{{MLConvTranspose2dOptions/groups}}
+        1. If |options|.{{MLConvTranspose2dOptions/outputSizes}} [=map/exists=], let |outputSizes| be |options|.{{MLConvTranspose2dOptions/outputSizes}}.
+        1. Else let |outputSizes| be the result of [=MLGraphBuilder/calculating convtranspose2d output sizes=] given |inputHeight|, |inputWidth|, |filterHeight|, |filterWidth|, |options|.{{MLConvTranspose2dOptions/padding}}, |options|.{{MLConvTranspose2dOptions/strides}}, |options|.{{MLConvTranspose2dOptions/dilations}}, and |options|.{{MLConvTranspose2dOptions/outputPadding}}.
         1. Switch on |options|.{{MLConvTranspose2dOptions/inputLayout}}:
             <dl class=switch>
                 : {{MLInputOperandLayout/"nchw"}}
-                :: Let |outputShape| be « |batches|, |channels|, floor( |outputSizes|[0] ), floor( |outputSizes|[1] ) ».
+                :: Let |outputShape| be « |batches|, |outputChannels|, floor( |outputSizes|[0] ), floor( |outputSizes|[1] ) ».
                 : {{MLInputOperandLayout/"nhwc"}}
-                :: Let |outputShape| be « |batches|, floor( |outputSizes|[0] ), floor( |outputSizes|[1] ), |channels| ».
+                :: Let |outputShape| be « |batches|, floor( |outputSizes|[0] ), floor( |outputSizes|[1] ), |outputChannels| ».
             </dl>
-    1. If |outputShape| is not the same as the shape of |options|.{{MLConvTranspose2dOptions/bias}}'s [=MLOperand/shape=], then [=exception/throw=] a {{TypeError}}.
+    1. If |inputChannels| is not equal to |filterInputChannels|, then [=exception/throw=] a {{TypeError}}.
```

Comment:
The steps between here and _"Make graph connections"_ seem like they're part of the _"Calculate the output shape"_ substeps and should be indented. (I don't know why I didn't have them indented previously.)

Reply:
These steps may use variables that are created in "Calculate the output shape" subsptes.

Reply:
Agreed - I'm not suggesting to move them, just indent them. I guess I'm thinking of the indented steps just being labeled blocks, but not scopes.

Reply:
I get your idea and agree. Fixed in https://github.com/webmachinelearning/webnn/pull/608/commits/98482cf2b65efb4b449d0e61cc2977d78468f01a.